test/OptionalObjectiveTest.js

```diff
@@ -0,0 +1,165 @@
+/*global require*/
+/*global describe*/
+/*global it*/
+
+var assert = require("assert");
+var solver = require("../src/main.js");
+
+describe("Testing if optional objectives are taken into account", function () {
+
+	it("should be able to construct and solve UI model", function () {
+
+		// Elements composing the UI
+		var nUIElements = 0;
+		function UIElement(model){
+			this.varULCornerX = model.addVariable(0, "varULCornerX"+nUIElements, false, false);
+			this.varULCornerY = model.addVariable(0, "varULCornerY"+nUIElements, false, false);
+
+			this.varWidth = model.addVariable(0, "varWidth"+nUIElements, false, false);
+			this.varHeight = model.addVariable(0, "varHeight"+nUIElements, false, false);
```

Comment:
Here I would replace ```varULCornerX```, ```varULCornerY```, ```varWidth``` and ```varHeight``` by x, y, w and h (including for the variable ids, e.g ```"varULCornerX"+nUIElements``` would simply be ```"x" + nUIElements```)